app/scripts/services/secrets.js

```diff
@@ -152,7 +152,7 @@ angular.module("openshiftConsole")
       // to list Secrets
       if (_.get(secret, 'secretReference.name') && webhookSecrets) {
         var matchingSecret = _.find(webhookSecrets, {metadata:{name: secret.secretReference.name}});
-        return decodeSecretData(matchingSecret.data).WebHookSecretKey;
+        return _.has(matchingSecret, 'data') ? decodeSecretData(matchingSecret.data).WebHookSecretKey : '';
```

Comment:
Prefer `_.get` so we never try to decode `null` values. (`_.has` only checks for `undefined`.)

```suggestion
        return _.get(matchingSecret, 'data') ? decodeSecretData(matchingSecret.data).WebHookSecretKey : '';
```